src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/RemoteAppAuthenticationHttpHandler.Framework.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Security.Claims;
+using System.Web;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+/// <summary>
+/// HTTP handler for serving requests to remote authenticaiton endpoint.
+/// </summary>
+internal sealed class RemoteAppAuthenticationHttpHandler : IHttpHandler
```

Comment:
we could implement `IHttpHandler` and `IHttpModule` together for this to keep the code together. If not, could the API key check be moved to the module? That way it fails early

Reply:
Combining them would work in most cases, but I worry about edge cases where some module after this one interacts with auth. I think it would be better to leave them separate but make sure all API key stuff happens in the module.